src/UrlParser.php

```diff
@@ -97,9 +102,20 @@ public function removeQueryStringCallback(callable $callback, string|null $url =
     protected function prepareUrl(string|null $url = null): string
     {
         if (null === $url) {
-            $url = Environment::get('requestUri');
+            if (null !== $this->requestStack) {
+                $url = $this->requestStack->getCurrentRequest()?->getUri();
+            } else {
+                // Fallback for manually created UrlParser instances
```

Comment:
was this ever a valid use case? Even if a service has no constructor arguments, it is not a BC promise to not change them?